tests/unit/test_table.py

```diff
@@ -1410,7 +1412,11 @@ def test___repr__(self):
         )
         self.assertEqual(repr(table1), expected)
 
-
+    def test___str__(self):
+        dataset = DatasetReference("project1", "dataset1")
```

Comment:
I would follow the convention in the other tests and name this var `dataset_ref`